app/src/main/java/io/homeassistant/companion/android/settings/language/LanguagesProvider.kt

```diff
@@ -0,0 +1,71 @@
+package io.homeassistant.companion.android.settings.language
+
+import android.content.Context
+import android.content.res.Resources
+import android.os.Build
+import android.os.LocaleList
+import android.util.DisplayMetrics
+import io.homeassistant.companion.android.BuildConfig
+import io.homeassistant.companion.android.R
+import java.util.Locale
+
+class LanguagesProvider {
+
+    fun getSupportedLanguages(context: Context): Map<String, String> {
+        val listAppLocales = mutableMapOf<String, String>()
+        val langManager = LanguagesManagerProvider().getManager(context)
+        val resources = context.resources
+
+        if (langManager.getAppVersion() != BuildConfig.VERSION_NAME || langManager.getLocales().isNullOrEmpty()) {
+            val listLocales = resources.assets.locales
+            val defString = getStringResource(context, "")
+            var supportedLocales = ""
+
+            listLocales.forEach {
+                if (getStringResource(context, it) != defString || it == Locale.ENGLISH.language) {
+                    val name = makeLocale(it).displayLanguage.capitalize()
+                    listAppLocales["$name ($it)"] = it
+                    supportedLocales += "$it,"
+                }
+            }
+            langManager.saveAppVersion(BuildConfig.VERSION_NAME)
+            langManager.saveLocales(supportedLocales)
+        } else {
+            val listLocales = langManager.getLocales()!!.split(',')
+            listLocales.forEach {
+                if (it.isNotEmpty()) {
+                    val name = makeLocale(it).displayLanguage.capitalize()
+                    listAppLocales["$name ($it)"] = it
+                }
+            }
+        }
+
+        listAppLocales[resources.getString(R.string.lang_option_label_default)] = resources.getString(R.string.lang_option_value_default)
```

Comment:
We should probably make the default option be the first option the user see's currently it is the last option.

Reply:
Agree, correct it